pkgs/stdenv/generic/make-derivation.nix

```diff
@@ -253,6 +253,9 @@ in rec {
           enableParallelChecking = attrs.enableParallelChecking or true;
         } // lib.optionalAttrs (hardeningDisable != [] || hardeningEnable != []) {
           NIX_HARDENING_ENABLE = enabledHardeningOptions;
+        } // lib.optionalAttrs (stdenv.hostPlatform.isx86_64 && stdenv.hostPlatform.platform.gcc.arch or stdenv.hostPlatform.parsed.cpu.arch or "x86-64" != "x86-64") {
+          requiredSystemFeatures = attrs.requiredSystemFeatures or [] ++ [ "gccarch-${stdenv.hostPlatform.platform.gcc.arch or stdenv.hostPlatform.parsed.cpu.arch}" ];
```

Comment:
Probably don’t want to use parsed.cpu.arch here. This should be part of the system tuple which is already handled by nix

Reply:
Instead i think we should only set this when the march flag is different than the triple’s specified cpu

Reply:
Yes, ```stdenv.hostPlatform.parsed.cpu.arch``` is always "x86-64" so it could be simplified